platform/ios/src/MGLMapView.mm

```diff
@@ -5830,38 +5844,38 @@ - (void)didUpdateLocationWithUserTrackingAnimated:(BOOL)animated completionHandl
         if (self.userTrackingState != MGLUserTrackingStateBegan)
         {
             // Keep both the user and the destination in view.
-            [self didUpdateLocationWithTargetAnimated:animated completionHandler:completion];
+            [self didUpdateLocationWithTargetDirection:direction animated:animated completionHandler:completion];
```

Comment:
Rather than changing all of these methods to have the direction parameter, i think it's better to make `directionByFollowingWithCourse` a readwrite property, and use it as previous in these methods.

Reply:
`directionByFollowingWithCourse` is also a computed property.